src/main/scala/temple/generate/language/service/go/GoServiceGenerator.scala

```diff
@@ -96,22 +106,71 @@ object GoServiceGenerator extends ServiceGenerator {
     sb.toString
   }
 
+  private def generateJsonMiddleware(): String =
+    readFile("src/main/scala/temple/generate/language/service/go/genFiles/JsonMiddleware.go")
+
+  private def generateHandler(serviceName: String, endpoint: Endpoint): String =
+    s"func ${serviceName + endpoint}Handler(w http.ResponseWriter, r *http.Request) {}\n\n"
+
+  private def generateHandlers(serviceName: String, endpoints: Set[Endpoint]): String = {
+    val sb = new StringBuilder
+    if (endpoints.contains(Endpoint.ReadAll)) {
+      sb.append(generateHandler(serviceName, Endpoint.ReadAll))
+    }
+    if (endpoints.contains(Endpoint.Create)) {
+      sb.append(generateHandler(serviceName, Endpoint.Create))
+    }
+    if (endpoints.contains(Endpoint.Read)) {
+      sb.append(generateHandler(serviceName, Endpoint.Read))
+    }
+    if (endpoints.contains(Endpoint.Update)) {
+      sb.append(generateHandler(serviceName, Endpoint.Update))
+    }
+    if (endpoints.contains(Endpoint.Delete)) {
+      sb.append(generateHandler(serviceName, Endpoint.Delete))
+    }
+    sb.toString
+  }
+
+  private def generateErrors(serviceName: String): String =
+    Seq(
+      "package dao",
+      "",
+      """import "fmt"""",
+      "",
+      s"// Err${serviceName.capitalize}NotFound is returned when a ${serviceName} for the provided ID was not found",
+      s"type Err${serviceName.capitalize}NotFound int64",
+      "",
+      s"func (e Err${serviceName.capitalize}NotFound) Error() string " + CodeWrap.curly
```

Comment:
personally this feels like a bit of an odd mix and match between interpolation and concatenation

Reply:
I agree, though I'm not sure I much prefer what I've changed it to, thoughts?

Reply:
That’s still a mix of concat and interpolation, just with slightly more long-winded concatenation

Reply:
Maybe reproduces the structure slightly artificially in the code, but how does this look?
```scala
errors +=
  s"func (e Err${serviceName.capitalize}NotFound) Error() string ${CodeWrap.curly.tabbed(
    s"""return fmt.Sprintf("${serviceName} not found with ID %d", e)""",
  )}"
```

Reply:
If you're happy with that lol

Reply:
I feel it’s nicer than the weird way that `CodeWrap.curly.tabbed` was being line-broken before, but up to you